aptos-move/block-executor/src/view.rs

```diff
@@ -1225,9 +1168,9 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<
                 for data_read in inner_reads.values() {
                     if let DataRead::Versioned(_version, value, Some(layout)) = data_read {
                         if let Some(bytes) = value.bytes() {
-                            let identifiers_in_read = self
-                                .extract_identifiers_from_value(bytes, layout.as_ref())
-                                .unwrap();
+                            let identifiers_in_read =
+                                extract_identifiers_from_value::<T>(bytes, layout.as_ref())
```

Comment:
Actually, a good question for all of us - currently things like replacement errors are code_invariant_errors, i.e. afaik Panic errors, so it wouldn't actually fallback to sequential. @igor-aptos am I right? Should we make a replacement error fall back to sequential? I can see arguments for both sides. 

For resource group serialization for instance, the fallback would be so we can skip the transaction, and that makes sense to me. Currently we also fallback on some resource group invariant errors, i.e. we finalized empty set but metadata does not say deletion. Maybe this should be treated the same way as say ID replacement, or smt that can be caused by MV complexity? The argument for is obviously that fallback might correct the issue, but the argument against is that maybe we don't want to keep going if there is some code invariant issue.. The fact that some things might be caused by concurrency and only on say 1 validator adds some extra points to fallback.

Thoughts?

Reply:
we do fallback to sequential if parallel has code invariant errors. 

Sequential fails if it has either speculative of code invariant errors. we should see if we can sometimes skip transactions there instead of failing